### PR TITLE
feat: support authorization record export

### PR DESCRIPTION
**Pull request by Ben-Sheppard**
## Description

This PR introduces the support in the new exporters for the `AuthorizationRecord` type along with tests to verify the exporting.

### CODE REVIEW

File: zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/entities/AuthorizationEntity.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+  public Map<PermissionType, List<String>> getPermissionValues() {
+    return permissionValues == null ? Map.of() : permissionValues;
+  }
+
+  public AuthorizationEntity setPermissionValues(
+      final Map<PermissionType, List<String>> permissionValues) {
+    this.permissionValues = permissionValues;
+    return this;
+  }
```
Comment (Ben-Sheppard) [inline zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/entities/AuthorizationEntity.java:71]: ⚠️ One thing I noticed as part of implementing tests is the ser/deser of this object coming back out of the indexes, I decided to flatten the entity field into concrete objects, this produces a document in ES with the same structure, for example:

```
        "hits": [
            {
                "_index": "authorizations",
                "_id": "2251799813685250",
                "_score": 1.0,
                "_source": {
                    "id": "2251799813685250",
                    "ownerKey": 2251799813685249,
                    "ownerType": "USER",
                    "resourceType": "DECISION_DEFINITION",
                    "permissionValues": {
                        "READ": [
                            "resourceTwo"
                        ],
                        "CREATE": [
                            "resourceOne"
                        ]
                    }
                }
            }
        ]
```

File: zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/entities/AuthorizationEntity.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+package io.camunda.exporter.entities;
+
+import io.camunda.zeebe.protocol.record.value.AuthorizationOwnerType;
+import io.camunda.zeebe.protocol.record.value.AuthorizationResourceType;
+import io.camunda.zeebe.protocol.record.value.PermissionType;
+import java.util.List;
+import java.util.Map;
+
+public class AuthorizationEntity implements ExporterEntity<AuthorizationEntity> {
+  private String id;
```
Comment (remcowesterhoud) [inline zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/entities/AuthorizationEntity.java:17]: ❓What will id be here? Where does it come from?

Comment (remcowesterhoud) [reply zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/entities/AuthorizationEntity.java:17]: Looking at the rest of the code I guess it's the record key

**Review by remcowesterhoud — APPROVED** [review]
Lgtm!

**Review by maryarm — APPROVED** [review]
💯

File: zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/entities/AuthorizationEntity.java (hunk 2 above)
Comment (Ben-Sheppard) [reply zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/entities/AuthorizationEntity.java:17]: Thats correct, at the moment its record key

**Pull request merged**